File: 🟢 Easy/2490 - Circular Sentence/_2490CircularSentence.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        words = sentence.split(" ")
        length = len(words)

        prev = words[length - 1][-1]

        for i in range(length):
            if words[i][0] != prev:
                return False
            prev = words[i][-1]

        return True

File: 🟢 Easy/2490 - Circular Sentence/_2490CircularSentence.c
bool isCircularSentence(char* sentence) {
    int n = strlen(sentence);

    if (sentence[0] != sentence[n - 1]) {
        return false;
    }

    for (int i = 1; i < n - 1; i++) {
        if (sentence[i] == ' ') {
            if (sentence[i - 1] != sentence[i + 1]) {
                return false;
            }
        }
    }

    return true;
}